service/controller/cloudconfig/cloud_config.go

```diff
@@ -88,19 +90,19 @@ func New(config Config) (*CloudConfig, error) {
 	return c, nil
 }
 
-func (c CloudConfig) getEncryptionkey(customObject providerv1alpha1.AzureConfig) (string, error) {
-	cluster, err := c.randomkeysSearcher.SearchCluster(key.ClusterID(&customObject))
+func (c CloudConfig) getEncryptionkey(ctx context.Context, customObject providerv1alpha1.AzureConfig) (string, error) {
+	cluster, err := c.randomkeysSearcher.SearchCluster(ctx, key.ClusterID(&customObject))
 	if err != nil {
 		return "", microerror.Mask(err)
 	}
 	return string(cluster.APIServerEncryptionKey), nil
 }
 
 func newCloudConfig(template string, params k8scloudconfig.Params) (string, error) {
-	c := k8scloudconfig.DefaultCloudConfigConfig()
-	c.Params = params
-	c.Template = template
-
+	c := k8scloudconfig.CloudConfigConfig{
+		Params:   params,
+		Template: template,
+	}
```

Comment:
Integration tests are all failing. I think it might be due to cloudconfig changes.

Looks like this was changed in https://github.com/giantswarm/k8scloudconfig/pull/776

Error in azure-operator logs from https://app.circleci.com/pipelines/github/giantswarm/azure-operator/1919/workflows/b0d84d92-218c-4dca-9a03-1550f94885a9/jobs/14588/steps

```
{"caller":"github.com/giantswarm/operatorkit/v2/pkg/resource/wrapper/retryresource/crud_resource.go:89","controller":"azure-operator-azureconfig-controller","event":"update","function":"GetDesiredState","level":"warning","loop":"6","message":"retrying due to error","object":"/apis/provider.giantswarm.io/v1alpha1/namespaces/default/azureconfigs/ci-wip-bfb6c-f9583","resource":"blobobject","stack":{"annotation":"template.CloudConfigConfig.validation error: component `KubernetesAPIHealthz` requires version following constraint `>= 0.1.1`, got `0.1.0`","kind":"invalidConfigError","stack":[{"file":"/go/pkg/mod/github.com/giantswarm/k8scloudconfig/v8@v8.0.0/pkg/template/cloudconfig.go","line":36},{"file":"/root/project/service/controller/cloudconfig/cloud_config.go","line":108},{"file":"/root/project/service/controller/resource/blobobject/desired.go","line":144},{"file":"/go/pkg/mod/github.com/giantswarm/operatorkit/v2@v2.0.0/pkg/resource/wrapper/retryresource/crud_resource.go","line":82}]},"time":"2020-08-12T10:13:10.102568+00:00","version":"1796"}
```

Reply:
Our test magicians (Jose and Chris) are on vacation, but someone of the rest of us mere mortals will take a look :smile:

Reply:
Ok found something, let's see if this fixes it https://github.com/giantswarm/azure-operator/pull/940/commits/3c6e9b7535ad19d173e9a2268d662ab5f8ab67b9